build/webpack.common.config.js

```diff
@@ -52,43 +80,21 @@ module.exports = {
 		rules: [
 			// typescript
 			{
-				test: /\.ts?$/, // Another convention is to use the .es6 filetype, but you then
-				// have to supply that explicitly in import statements, which isn't cool.
+				test: /\.ts?$/,
 				exclude: [/(node_modules|bower_components)/],
-				loader: 'ts-loader'
+				use: [
+					babelLoaderConfig,
+					{
+						loader: 'ts-loader'
```

Comment:
Because TS is based on Babel do we potentially have a double-Babel situation going on here... Is the intention for TS loader here to output `esnext` code (i.e. just strip the TS annotations) and the inherited Babel loader to do the rest? And if so, should any expected`tsconfig.json` rules be defined here?

Reply:
According to the `ts-loader` docs, there is no danger: [example usage with babel](https://github.com/TypeStrong/ts-loader/blob/master/examples/react-babel-karma-gulp/webpack.config.base.js)

`n-ui` need be opinionated only on the client-side js it bundles, but we could have a common `tsconfig.json` in `n-gage`.